server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManager.java

```diff
@@ -74,8 +76,13 @@ public class RemoteClusterStateCleanupManager implements Closeable {
     private long lastCleanupAttemptStateVersion;
     private final ThreadPool threadpool;
     private final ClusterApplierService clusterApplierService;
+    private final Optional<RemoteRoutingTableService> remoteRoutingTableService;
 
-    public RemoteClusterStateCleanupManager(RemoteClusterStateService remoteClusterStateService, ClusterService clusterService) {
+    public RemoteClusterStateCleanupManager(
+        RemoteClusterStateService remoteClusterStateService,
+        ClusterService clusterService,
+        Optional<RemoteRoutingTableService> remoteRoutingTableService
```

Comment:
Use the factory implementation used in the https://github.com/opensearch-project/OpenSearch/pull/13870 once that is merged

Reply:
Addressed in next revision.